pkg/build/controller/factory/factory.go

```diff
@@ -68,110 +65,6 @@ func limitedLogAndRetry(buildupdater buildclient.BuildUpdater, maxTimeout time.D
 	}
 }
 
-// BuildControllerFactory constructs BuildController objects
-type BuildControllerFactory struct {
-	OSClient            osclient.Interface
-	KubeClient          kclientset.Interface
-	ExternalKubeClient  kclientsetexternal.Interface
-	BuildUpdater        buildclient.BuildUpdater
-	BuildLister         buildclient.BuildLister
-	BuildConfigGetter   buildclient.BuildConfigGetter
-	BuildDeleter        buildclient.BuildDeleter
-	DockerBuildStrategy *strategy.DockerBuildStrategy
-	SourceBuildStrategy *strategy.SourceBuildStrategy
-	CustomBuildStrategy *strategy.CustomBuildStrategy
-	BuildDefaults       builddefaults.BuildDefaults
-	BuildOverrides      buildoverrides.BuildOverrides
-
-	// Stop may be set to allow controllers created by this factory to be terminated.
-	Stop <-chan struct{}
-}
-
-// Create constructs a BuildController
-func (factory *BuildControllerFactory) Create() controller.RunnableController {
-	queue := cache.NewResyncableFIFO(cache.MetaNamespaceKeyFunc)
-	cache.NewReflector(newBuildLW(factory.OSClient), &buildapi.Build{}, queue, 2*time.Minute).RunUntil(factory.Stop)
-
-	eventBroadcaster := record.NewBroadcaster()
-	eventBroadcaster.StartRecordingToSink(&kv1core.EventSinkImpl{Interface: kv1core.New(factory.ExternalKubeClient.CoreV1().RESTClient()).Events("")})
-
-	client := ControllerClient{factory.KubeClient, factory.OSClient}
-	buildController := &buildcontroller.BuildController{
-		BuildUpdater:      factory.BuildUpdater,
-		BuildLister:       factory.BuildLister,
-		BuildConfigGetter: factory.BuildConfigGetter,
-		BuildDeleter:      factory.BuildDeleter,
-		ImageStreamClient: client,
-		PodManager:        client,
-		RunPolicies:       policy.GetAllRunPolicies(factory.BuildLister, factory.BuildUpdater),
-		BuildStrategy: &typeBasedFactoryStrategy{
-			DockerBuildStrategy: factory.DockerBuildStrategy,
-			SourceBuildStrategy: factory.SourceBuildStrategy,
-			CustomBuildStrategy: factory.CustomBuildStrategy,
-		},
-		Recorder:       eventBroadcaster.NewRecorder(kapi.Scheme, kclientv1.EventSource{Component: "build-controller"}),
-		BuildDefaults:  factory.BuildDefaults,
-		BuildOverrides: factory.BuildOverrides,
-	}
-
-	return &controller.RetryController{
-		Queue: queue,
-		RetryManager: controller.NewQueueRetryManager(
-			queue,
-			cache.MetaNamespaceKeyFunc,
-			limitedLogAndRetry(factory.BuildUpdater, 30*time.Minute),
```

Comment:
possibly can be more aggressive with code deletion here? for example, I think maybe limitedLogAndRetry, TestLimitedLogAndRetryFinish and TestLimitedLogAndRetryProcessing can be dispatched.  Check if there are more.  I'll be sorry to see `errors.New("funky error")` go though :)